src/mats_l1_processing/photometer.py

```diff
@@ -0,0 +1,93 @@
+#%%
+import numpy as np
+from mats_l1_processing.instrument import Photometer
+from pandas import DataFrame
+
+def calibrate_pm(df: DataFrame, photometer: Photometer):
+    
+# ===============================================================================================================
+# Calibration starts here
+# PM1 is Bkg photometer, 754BP3 nm / PM2 is A-band photometer, 763BP8 nm
+#   Change to bits/s
+    PM1_Sig_bit = df['PM1S']/df['PM1SCNTR']
+    PM1_Tpd_bit = df['PM1A']/df['PM1ACNTR']
+    PM1_Tif_bit = df['PM1B']/df['PM1BCNTR']
+    PM2_Sig_bit = df['PM2S']/df['PM2SCNTR']
+    PM2_Tpd_bit = df['PM2A']/df['PM2ACNTR']
+    PM2_Tif_bit = df['PM2B']/df['PM2BCNTR']
+    pm_texp = round(df['PM1SCNTR']/166) # exposure time in seconds, 1 second "integration" is 166-167 sampling cycles
+
+# ===========================================================
+# Calibrate data
+
+#   extracting variables from dict
+    bitar = photometer.cal_therm['bitar'] # 0.1 to 4095 bits in steps of 0.1 bit, array of float64 (1,40950)
+    TempFM1if_raw = photometer.cal_therm['TempFM1if_raw'] # array of float64 (1,40950)
+    TempFM1pd_raw = photometer.cal_therm['TempFM1pd_raw'] # array of float64 (1,40950)
+    TempFM2if_raw = photometer.cal_therm['TempFM2if_raw'] # array of float64 (1,40950)
+    TempFM2pd_raw = photometer.cal_therm['TempFM2pd_raw'] # array of float64 (1,40950)
+    Temperatur = photometer.cal_rad['Temperatur'] # -20 to +44 °C in steps of 0.1 °C,  # array of float64 (1,641)
+    SignFM1_Rad_raw = photometer.cal_rad['SignFM1_Rad_raw'] # 2D matrix, array of float64 (641,40950)
+    SignFM2_Rad_raw = photometer.cal_rad['SignFM2_Rad_raw'] # 2D matrix, array of float64 (641,40950)
+
+# =================================
+# Change raw temperature data to °C
+#   define pmBkg_Tpd, pmBkg_Tif, pmAband_Tpd, pmAband_Tif
+    pmBkg_Tpd = np.NaN * np.ones_like(PM1_Tpd_bit)   # Temperature of the Bkg photometer photodiode
+    pmBkg_Tif = np.NaN * np.ones_like(PM1_Tif_bit)   # Temperature of the Bkg photometer filter
+    pmAband_Tpd = np.NaN * np.ones_like(PM2_Tpd_bit) # Temperature of the A-band photometer photodiode
+    pmAband_Tif = np.NaN * np.ones_like(PM2_Tif_bit) # Temperature of the A-band photometer filter
+
+    for ij in range(0, len(PM1_Tpd_bit)-1):
```

Comment:
Same comment regarding `-1` as below.
```suggestion
    for ij in range(len(PM1_Tpd_bit)):
```

Reply:
fixed